### PR TITLE
standardize tags param

### PR DESCRIPTION
**Pull request by halberom**
Most of the new ec2 modules use `tags:` as the tags parameter name.  However several of the core modules use `snapshot_tags`, `instance_tags`, `ami_tags` etc.

Updating them so they all accept `tags:`, and use aliases for backward compatibility.

Tested all 3 with before and after tag names.  See [here](https://gist.github.com/halberom/9a440ae21e88ff18fd19) for testing output

I've not updated the ec2_vpc module, because the new ones (ec2_vpc_net etc) replace it, so I'm assuming it's deprecated although I can do that too if desired.

### CODE REVIEW

**Comment by halberom** [comment]
I could probably have just added 'tags' as the alias, instead of the find/replace I did, but again, all the other modules use `tags` in the code too.

**Comment by gregdek** [comment]
Since this affects multiple AWS modules, pinging all the AWS folks here for review:

@jsmartin @scicoin-project @tombamford @garethr @lorin @jarv @jsdalton @silviud @adq @zbal @zeekin @willthames @lwade @carsongee @defionscode @tastychutney @bpennypacker @loia

Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: cloud/amazon/ec2.py [hunk 1]
```diff
@@ -229,17 +229,17 @@
...
     aliases: []
-  count_tag:
+  count_tags:
     version_added: "1.5"
     description:
       - Used with 'exact_count' to determine how many nodes based on a specific tag criteria should be running.  This can be expressed in multiple ways and is shown in the EXAMPLES section.  For instance, one can request 25 servers that are tagged with "class=webserver".
     required: false
     default: null
-    aliases: []
+    aliases: ['count_tags']
```
Comment (manicminer) [inline cloud/amazon/ec2.py]: Should the alias here perhaps be `count_tag` (i.e. singular) like the original arg?

needs_revision ?

**Comment by jarv** [comment]
shipit

**Comment by halberom** [comment]
good catch, thanks Tom.

**Comment by manicminer** [comment]
No problem! This is a sensible change. shipit

**Comment by gregdek** [comment]
Thanks for the quick review gents, and thanks to @halberom for the PR.

Marking for inclusion. cc: @ansible/core team.

**Comment by juliedavila** [comment]
+1

**Comment by bcoca** [comment]
version_added needs to be updated for all. I would suggest instead to make 'tags' as an alias and make a comment 'tags alias added in 2.0'

**Comment by juliedavila** [comment]
@halberom friendly ping

**Comment by halberom** [comment]
Busy with other stuff atm, will redo at a later date when have time.

**Pull request closed without merging**